### ISSUE TITLE
[JENKINS-4804] exceptions when submitting form in 1.333 with Winstone

### ISSUE THREAD

**Issue by timja** [timja/jenkins-gh-issues-poc-06-20#295]
<p>I get the following exceptions both in IE and chrome when submitting a configuration form using 1.333, using 1.330 this works fine.</p>

<p><span class="error">&#91;Winstone 2009/11/09 23:22:02&#93;</span> - Content-length said 8154, actual length was 1460<br/>
<span class="error">&#91;webapp 2009/11/09 23:22:02&#93;</span> - Error while serving <a href="http://[hostname" class="external-link" target="_blank" rel="nofollow noopener">http://[hostname</a> deleted]null<br/>
java.lang.reflect.InvocationTargetException<br/>
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)<br/>
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
at java.lang.reflect.Method.invoke(Method.java:597)<br/>
at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:185)<br/>
at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:101)<br/>
at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:54)<br/>
at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:74)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:180)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:409)<br/>
at org.kohsuke.stapler.Stapler.service(Stapler.java:118)<br/>
at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)<br/>
at winstone.ServletConfiguration.execute(ServletConfiguration.java:249)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:335)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:378)<br/>
at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:94)<br/>
at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:86)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:47)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:166)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.basicauth.BasicProcessingFilter.doFilter(BasicProcessingFilter.java:173)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)<br/>
at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:66)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:333)<br/>
at winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:244)<br/>
at winstone.RequestHandlerThread.run(RequestHandlerThread.java:150)<br/>
at java.lang.Thread.run(Thread.java:619)<br/>
Caused by: java.lang.Error: This page expects a form submission<br/>
at org.kohsuke.stapler.RequestImpl.getSubmittedForm(RequestImpl.java:686)<br/>
at hudson.model.Job.doConfigSubmit(Job.java:946)<br/>
at hudson.model.AbstractProject.doConfigSubmit(AbstractProject.java:519)<br/>
... 50 more</p>

<p><span class="error">&#91;Winstone 2009/11/09 23:22:02&#93;</span> - Untrapped Error in Servlet<br/>
javax.servlet.ServletException: java.lang.Error: This page expects a form submission<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:505)<br/>
at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:180)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:409)<br/>
at org.kohsuke.stapler.Stapler.service(Stapler.java:118)<br/>
at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)<br/>
at winstone.ServletConfiguration.execute(ServletConfiguration.java:249)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:335)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:378)<br/>
at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:94)<br/>
at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:86)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:47)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:166)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.basicauth.BasicProcessingFilter.doFilter(BasicProcessingFilter.java:173)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)<br/>
at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:66)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:333)<br/>
at winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:244)<br/>
at winstone.RequestHandlerThread.run(RequestHandlerThread.java:150)<br/>
at java.lang.Thread.run(Thread.java:619)<br/>
Caused by: java.lang.Error: This page expects a form submission<br/>
at org.kohsuke.stapler.RequestImpl.getSubmittedForm(RequestImpl.java:686)<br/>
at hudson.model.Job.doConfigSubmit(Job.java:946)<br/>
at hudson.model.AbstractProject.doConfigSubmit(AbstractProject.java:519)<br/>
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)<br/>
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
at java.lang.reflect.Method.invoke(Method.java:597)<br/>
at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:185)<br/>
at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:101)<br/>
at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:54)<br/>
at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:74)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
... 40 more</p>

<p><span class="error">&#91;Winstone 2009/11/09 23:22:02&#93;</span> - Error within request handler thread</p>

<p>winstone.WinstoneException: Error URI Line: t_artifacts=on&<em>.artifacts=&</em>.targetPlatformName=&deploy.war=&stapler-<br/>
class=hudson.plugins.deploy.Tomcat6xAdapter&deploy.userName=&deploy.password=&deploy.url=&stapler-<br/>
class=hudson.plugins.deploy.Tomcat5xAdapter&deploy.userName=&deploy.password=&deploy.url=&stapler-<br/>
class=hudson.plugins.deploy.Tomcat4xAdapter&deploy.userName=&deploy.password=&deploy.url=&stapler-<br/>
class=hudson.plugins.deploy.JBoss5xAdapter&deploy.userName=&deploy.password=&deploy.url=&stapler-<br/>
class=hudson.plugins.deploy.JBoss4xAdapter&deploy.userName=&deploy.password=&deploy.url=&stapler-<br/>
class=hudson.plugins.deploy.JBoss3xAdapter&deploy.userName=&deploy.password=&deploy.url=&pattern=&healthy=&unHealthy=&threshold=&newThre<br/>
shold=&failureThreshold=&newFailureThreshold=&checkstyle.thresholdLimit=low&_.defaultEncoding=&hudson-plugins-jira-<br/>
JiraIssueUpdater=on&hudson-plugins-warnings-<br/>
WarningsPublisher=on&canScanConsole=on&pattern=&includePattern=&excludePattern=&healthy=&unHealthy=&threshold=&newThreshold=&failureThre<br/>
shold=&newFailureThreshold=&warnings.thresholdLimit=low&_.defaultEncoding=&hudson-plugins-tasks-<br/>
TasksPublisher=on&pattern=&excludePattern=&high=FIXME%2C+HACK&normal=TODO&low=@deprecated&healthy=&unHealthy=&threshold=&newThreshold=&f<br/>
ailureThreshold=&newFailureThreshold=&tasks.thresholdLimit=low&_.defaultEncoding=&hudson-plugins-emailext-<br/>
ExtendedEmailPublisher=on&recipientlist_recipients=kooper@ncsa.uiuc.edu%2C+pbajcsy@ncsa.uiuc.edu&project_default_subject=%24DEFAULT_SUBJ<br/>
ECT&project_default_content=%24DEFAULT_CONTENT&mailer.Failure.configured=true&mailer.Failure.sendToRecipientList=on&mailer.Failure.sendT<br/>
oDevelopers=on&mailer.Failure.includeCulprits=on&mailer.Failure.recipientList=&mailer.Failure.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.<br/>
Failure.body=%24PROJECT_DEFAULT_CONTENT&mailer.Success.configured=true&mailer.Success.sendToRecipientList=on&mailer.Success.sendToDevelo<br/>
pers=on&mailer.Success.recipientList=&mailer.Success.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.Success.body=%24PROJECT_DEFAULT_CONTENT&m<br/>
ailer.Unstable.configured=false&mailer.Unstable.sendToRecipientList=on&mailer.Unstable.sendToDevelopers=on&mailer.Unstable.recipientList<br/>
=&mailer.Unstable.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.Unstable.body=%24PROJECT_DEFAULT_CONTENT&mailer.Still-<br/>
Failing.configured=false&mailer.Still-Failing.sendToDevelopers=on&mailer.Still-Failing.recipientList=&mailer.Still-<br/>
Failing.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.Still-<br/>
Failing.body=%24PROJECT_DEFAULT_CONTENT&mailer.Fixed.configured=false&mailer.Fixed.sendToRecipientList=on&mailer.Fixed.sendToDevelopers=<br/>
on&mailer.Fixed.recipientList=&mailer.Fixed.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.Fixed.body=%24PROJECT_DEFAULT_CONTENT&mailer.Still<br/>
<del>Unstable.configured=false&mailer.Still-Unstable.sendToDevelopers=on&mailer.Still-Unstable.recipientList=&mailer.Still</del><br/>
Unstable.subject=%24PROJECT_DEFAULT_SUBJECT&mailer.Still-<br/>
Unstable.body=%24PROJECT_DEFAULT_CONTENT&mailer_recipients=&mailer_notifyEveryUnstableBuild=on&json=%7B%22name%22%3A+%22Im2Learn%22%2C+%<br/>
22description%22%3A+%22%22%2C+%22logrotate%22%3A+%7B%22logrotate_days%22%3A+%22%22%2C+%22logrotate_nums%22%3A+%2210%22%7D%2C+%22jobPrope<br/>
rty0%22%3A+%7B%7D%2C+%22jobProperty1%22%3A+%7B%7D%2C+%22jdk%22%3A+%22%22%2C+%22scm%22%3A+%7B%22value%22%3A+%221%22%2C+%22cvsRoot%22%3A+%<br/>
22%2Fhome%2Fcvs%22%2C+%22allModules%22%3A+%22im2learn%22%2C+%22branch%22%3A+%22%22%2C+%22isTag%22%3A+false%2C+%22cvsRsh%22%3A+%22%22%2C+<br/>
%22legacy%22%3A+false%2C+%22canUseUpdate%22%3A+true%2C+%22%22%3A+%22auto%22%2C+%22excludedRegions%22%3A+%22%22%7D%2C+%22hudson-triggers-<br/>
SCMTrigger%22%3A+%7B%22scmpoll_spec%22%3A+%22%23+every+15+minutes%5Cr%5Cn*%2F15+<b><ins></b></ins><b><ins></b>%5Cr%5Cn%22%7D%2C</ins>%22builder%22%3A+%7B%22antName%<br/>
22%3A+%22ANT+1.7%22%2C+%22targets%22%3A+%22realclean+installer+cyberintegrator.jar+copyisda%22%2C+%22buildFile%22%3A+%22%22%2C+%22proper<br/>
ties%22%3A+%22%22%2C+%22antOpts%22%3A+%22%22%2C+%22stapler-<br/>
class%22%3A+%22hudson.tasks.Ant%22%2C+%22kind%22%3A+%22hudson.tasks.Ant%24DescriptorImpl%22%7D%2C+%22hudson-tasks-<br/>
ArtifactArchiver%22%3A+%7B%22artifacts%22%3A+%22build%2FIm2Learn_*%2Cbuild%2FIm2Learn.jar%2Cbui<br/>
at winstone.HttpListener.parseURILine(HttpListener.java:318)<br/>
at winstone.HttpListener.parseURI(HttpListener.java:246)<br/>
at winstone.RequestHandlerThread.run(RequestHandlerThread.java:87)<br/>
at java.lang.Thread.run(Thread.java:619)</p>

---
<details><summary><i>Originally reported by <a title="kooper" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kooper">kooper</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-4804" target="_blank">exceptions when submitting form in 1.333 with Winstone</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Kohsuke Kawaguchi" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-01-07T11:13:51+00:00
<li><b>imported</b>: 2022-06-20
</ul></i>
</details>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kooper">kooper</a>:</i>
<p>This could be related to the following exception in my log file:</p>

<p><span class="error">&#91;Winstone 2009/11/09 23:20:13&#93;</span> - Untrapped Error in Servlet<br/>
javax.servlet.ServletException: net.sf.json.JSONException: Unterminated string <br/>
at character 919 of </p>
{"connectionCheckUrl": "http://www.google.com/", "core": 
{"buildDate": "Nov 09, 2009", "name": "core", "url": "http://hudson-
ci.org/download/war/1.333/hudson.war", "version": "1.333"}
<p>, "id": "default", <br/>
"plugins": {"SCTMExecutor": {"buildDate": "Oct 18, 2009", "dependencies": [], <br/>
"developers": [</p>
{"developerId": "tofuAtjava", "email": 
"tfuerer.javanet@googlemail.com", "name": "Thomas Fuerer"}
<p>], "excerpt": "This <br/>
plugin will let users use Borland's SilkCentral Test Manager 2008 R2 or later.", <br/>
"name": "SCTMExecutor", "previousTimestamp": "2009-06-04T00:52:02.00Z", <br/>
"releaseTimestamp": "2009-10-19T00:16:10.00Z", "requiredCore": "1.323", "title": <br/>
"SCTMExecutor", "url": "http://hudson-<br/>
ci.org/download/plugins/SCTMExecutor/1.2/SCTMExecutor.hpi", "version": "1.2", <br/>
"wiki": "http://wiki.jenkins-ci.org//display/JENKINS/SCTMExecutor"}, "Schmant": <br/>
{"buildDate": "Apr 01, 2009", "dependencies": [], "developers": [{"developerId%<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:505)<br/>
	at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:180)<br/>
	at <br/>
org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
	at org.kohsuke.stapler.MetaClass$4.doDispatch(MetaClass.java:144)<br/>
	at <br/>
org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:409)<br/>
	at org.kohsuke.stapler.Stapler.service(Stapler.java:118)<br/>
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)<br/>
	at winstone.ServletConfiguration.execute(ServletConfiguration.java:249)<br/>
	at winstone.RequestDispatcher.forward(RequestDispatcher.java:335)<br/>
	at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:378)<br/>
	at <br/>
hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:94)<br/>
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:86)<br/>
	at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
	at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:47)<br/>
	at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
	at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
	at <br/>
hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFi<br/>
lter.java:51)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFil<br/>
ter.java:166)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(Anonymo<br/>
usProcessingFilter.java:125)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMePr<br/>
ocessingFilter.java:135)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.<br/>
java:271)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.ui.basicauth.BasicProcessingFilter.doFilter(BasicProcessingFil<br/>
ter.java:173)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessi<br/>
onContextIntegrationFilter.java:249)<br/>
	at <br/>
hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContext<br/>
IntegrationFilter2.java:66)<br/>
	at <br/>
hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
	at <br/>
hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
	at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)<br/>
	at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)<br/>
	at winstone.RequestDispatcher.forward(RequestDispatcher.java:333)<br/>
	at <br/>
winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:244)<br/>
	at winstone.RequestHandlerThread.run(RequestHandlerThread.java:150)<br/>
	at java.lang.Thread.run(Thread.java:619)<br/>
Caused by: net.sf.json.JSONException: Unterminated string at character 919 of <br/>
{"connectionCheckUrl": "http://www.google.com/", "core": </p>
{"buildDate": "Nov 09, 
2009", "name": "core", "url": "http://hudson-
ci.org/download/war/1.333/hudson.war", "version": "1.333"}
<p>, "id": "default", <br/>
"plugins": {"SCTMExecutor": {"buildDate": "Oct 18, 2009", "dependencies": [], <br/>
"developers": [</p>
{"developerId": "tofuAtjava", "email": 
"tfuerer.javanet@googlemail.com", "name": "Thomas Fuerer"}
<p>], "excerpt": "This <br/>
plugin will let users use Borland's SilkCentral Test Manager 2008 R2 or later.", <br/>
"name": "SCTMExecutor", "previousTimestamp": "2009-06-04T00:52:02.00Z", <br/>
"releaseTimestamp": "2009-10-19T00:16:10.00Z", "requiredCore": "1.323", "title": <br/>
"SCTMExecutor", "url": "http://hudson-<br/>
ci.org/download/plugins/SCTMExecutor/1.2/SCTMExecutor.hpi", "version": "1.2", <br/>
"wiki": "http://wiki.jenkins-ci.org//display/JENKINS/SCTMExecutor"}, "Schmant": <br/>
{"buildDate": "Apr 01, 2009", "dependencies": [], "developers": [{"developerId%<br/>
	at net.sf.json.util.JSONTokener.syntaxError(JSONTokener.java:512)<br/>
	at net.sf.json.util.JSONTokener.nextString(JSONTokener.java:244)<br/>
	at net.sf.json.util.JSONTokener.nextValue(JSONTokener.java:352)<br/>
	at net.sf.json.JSONObject._fromJSONTokener(JSONObject.java:834)<br/>
	at net.sf.json.JSONObject.fromObject(JSONObject.java:170)<br/>
	at net.sf.json.util.JSONTokener.nextValue(JSONTokener.java:355)<br/>
	at net.sf.json.JSONArray._fromJSONTokener(JSONArray.java:917)<br/>
	at net.sf.json.JSONArray.fromObject(JSONArray.java:139)<br/>
	at net.sf.json.util.JSONTokener.nextValue(JSONTokener.java:358)<br/>
	at net.sf.json.JSONObject._fromJSONTokener(JSONObject.java:850)<br/>
	at net.sf.json.JSONObject.fromObject(JSONObject.java:170)<br/>
	at net.sf.json.util.JSONTokener.nextValue(JSONTokener.java:355)<br/>
	at net.sf.json.JSONObject._fromJSONTokener(JSONObject.java:850)<br/>
	at net.sf.json.JSONObject.fromObject(JSONObject.java:170)<br/>
	at net.sf.json.util.JSONTokener.nextValue(JSONTokener.java:355)<br/>
	at net.sf.json.JSONObject._fromJSONTokener(JSONObject.java:850)<br/>
	at net.sf.json.JSONObject._fromString(JSONObject.java:1039)<br/>
	at net.sf.json.JSONObject.fromObject(JSONObject.java:176)<br/>
	at net.sf.json.JSONObject.fromObject(JSONObject.java:147)<br/>
	at hudson.model.UpdateSite.doPostBack(UpdateSite.java:136)<br/>
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>
	at <br/>
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)<br/>
	at <br/>
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.jav<br/>
a:25)<br/>
	at java.lang.reflect.Method.invoke(Method.java:597)<br/>
	at <br/>
org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:185)<br/>
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:101)<br/>
	at <br/>
org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:54)<br/>
	at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:74)<br/>
	at <br/>
org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:30)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:493)<br/>
	... 43 more</p>

<p><span class="error">&#91;Winstone 2009/11/09 23:20:13&#93;</span> - Error within request handler thread</p>

<p>winstone.WinstoneException: Error URI Line: <br/>
22%3A%20%22kalle%22%7D%5D%2C%20%22name%22%3A%20%22Schmant%22%2C%20%22releaseTime<br/>
stamp%22%3A%20%222009-04-<br/>
01T18%3A28%3A56.00Z%22%2C%20%22requiredCore%22%3A%20%221.295%22%2C%20%22title%22<br/>
%3A%20%22Schmant%20Plugin%22%2C%20%22url%22%3A%20%22http%3A%2F%2Fhudson-<br/>
ci.org%2Fdownload%2Fplugins%2FSchmant%2F1.0%2FSchmant.hpi%22%2C%20%22version%22%<br/>
3A%20%221.0%22%2C%20%22wiki%22%3A%20%22http%3A%2F%2Fwiki.hudson-<br/>
ci.org%2F%2Fdisplay%2FHUDSON%2FSchmant%2BPlugin%22%7D%2C%20%22URLSCM%22%3A%20%7B<br/>
%22buildDate%22%3A%20%22Jun%2024%2C%202008%22%2C%20%22dependencies%22%3A%20%5B%5<br/>
D%2C%20%22developers%22%3A%20%5B%7B%22developerId%22%3A%20%22mdonohue%22%7D%5D%2<br/>
C%20%22excerpt%22%3A%20%22This%20plugin%20allows%20the%20use%20of%20URLs%20as%20<br/>
an%20SCM%20plugin%22%2C%20%22name%22%3A%20%22URLSCM%22%2C%20%22previousTimestamp<br/>
%22%3A%20%222008-05-<br/>
16T06%3A08%3A54.00Z%22%2C%20%22releaseTimestamp%22%3A%20%222008-06-<br/>
25T04%3A13%3A30.00Z%22%2C%20%22requiredCore%22%3A%20%221.229%22%2C%20%22title%22<br/>
%3A%20%22URL%20SCM%22%2C%20%22url%22%3A%20%22http%3A%2F%2Fhudson-<br/>
ci.org%2Fdownload%2Fplugins%2FURLSCM%2F1.4%2FURLSCM.hpi%22%2C%20%22version%22%3A<br/>
%20%221.4%22%2C%20%22wiki%22%3A%20%22http%3A%2F%2Fwiki.hudson-<br/>
ci.org%2F%2Fdisplay%2FHUDSON%2FURL%2BSCM%22%7D%2C%20%22accurev%22%3A%20%7B%22bui<br/>
ldDate%22%3A%20%22Jul%2020%2C%202009%22%2C%20%22dependencies%22%3A%20%5B%5D%2C%2<br/>
0%22developers%22%3A%20%5B%7B%22developerId%22%3A%20%22statum%22%7D%5D%2C%20%22e<br/>
xcerpt%22%3A%20%22This%20plugin%20allows%20you%20to%20use%20%3Ca%20href%3D'http%<br/>
3A%2F%2Fwww.accurev.com%2F'%3EAccuRev%3C%2Fa%3E%20as%20a%20SCM.%22%2C%20%22name%<br/>
22%3A%20%22accurev%22%2C%20%22previousTimestamp%22%3A%20%222009-01-<br/>
30T20%3A39%3A50.00Z%22%2C%20%22releaseTimestamp%22%3A%20%222009-07-<br/>
20T07%3A30%3A20.00Z%22%2C%20%22requiredCore%22%3A%20%221.316%22%2C%20%22title%22<br/>
%3A%20%22Accurev%20Plugin%22%2C%20%22url%22%3A%20%22http%3A%2F%2Fhudson-<br/>
ci.org%2Fdownload%2Fplugins%2Faccurev%2F0.6.8%2Faccurev.hpi%22%2C%20%22version%2<br/>
2%3A%20%220.6.8%22%2C%20%22wiki%22%3A%20%22http%3A%2F%2Fwiki.hudson-<br/>
ci.org%2F%2Fdisplay%2FHUDSON%2FAccurev%2BPlugin%22%7D%2C%20%22active-<br/>
directory%22%3A%20%7B%22buildDate%22%3A%20%22Jul%2010%2C%202009%22%2C%20%22depen<br/>
dencies%22%3A%20%5B%5D%2C%20%22developers%22%3A%20%5B%7B%22developerId%22%3A%20%<br/>
22nordj%22%7D%5D%2C%20%22excerpt%22%3A%20%22With%20this%20plugin%2C%20you%20can%<br/>
20configure%20Hudson%20authenticates%20the%20username%20and%20the%20password%20t<br/>
hrough%20Active%20Directory.%20%22%2C%20%22name%22%3A%20%22active-<br/>
directory%22%2C%20%22previousTimestamp%22%3A%20%222009-07-<br/>
02T21%3A04%3A36.00Z%22%2C%20%22releaseTimestamp%22%3A%20%222009-07-<br/>
10T20%3A09%3A24.00Z%22%2C%20%22requiredCore%22%3A%20%221.314%22%2C%20%22title%22<br/>
%3A%20%22Active%20Directory%20plugin%22%2C%20%22url%22%3A%20%22http%3A%2F%2Fhuds<br/>
on-ci.org%2Fdownload%2Fplugins%2Factive-directory%2F1.15%2Factive-<br/>
directory.hpi%22%2C%20%22version%22%3A%20%221.15%22%2C%20%22wiki%22%3A%20%22http<br/>
%3A%2F%2Fwiki.hudson-<br/>
ci.org%2F%2Fdisplay%2FHUDSON%2FActive%2BDirectory%2Bplugin%22%7D%2C%20%22audit-<br/>
trail%22%3A%20%7B%22buildDate%22%3A%20%22Mar%2003%2C%202009%22%2C%20%22dependenc<br/>
ies%22%3A%20%5B%5D%2C%20%22developers%22%3A%20%5B%7B%22developerId%22%3A%20%22mi<br/>
ndless%22%7D%5D%2C%20%22excerpt%22%3A%20%22Keep%20a%20log%20of%20who%20performed<br/>
%20particular%20Hudson%20operations%2C%20such%20as%20configuring%20jobs.%22%2C%2<br/>
0%22name%22%3A%20%22audit-trail%22%2C%20%22previousTimestamp%22%3A%20%222009-02-<br/>
14T01%3A08%3A42.00Z%22%2C%20%22releaseTimestamp%22%3A%20%222009-03-<br/>
04T02%3A04%3A02.00Z%22%2C%20%22requiredCore%22%3A%20%221.288%22%2C%20%22title%22<br/>
%3A%20%22Audit%20Trail%20Plugin%22%2C%20%22url%22%3A%20%22http%3A%2F%2Fhudson-<br/>
ci.org%2Fdownload%2Fplugins%2Faudit-trail%2F1.3%2Faudit-<br/>
trail.hpi%22%2C%20%22version%22%3A%20%221.3%22%2C%20%22wiki%22%3A%20%22http%3A%2<br/>
F%2Fwiki.hudson-<br/>
ci.org%2F%2Fdisplay%2FHUDSON%2FAudit%2BTrail%2BPlugin%22%7D%2C%20%22backlog%22%3<br/>
A%20%7B%22buildDate%22%3A%20%22May%2026%2C%202009%22%2C%20%22dependencies%22%3A%<br/>
20%5B%5D%2C%20%22developers%22%3A%20%5B%7B%22developerId%22%3A%20%22yamamoto%22%<br/>
7D%5D%2C%20%22excerpt%22%3A%20%22This%20plugin%20integrates%20%3Ca%20href%3D'htt<br/>
p%3A<br/>
	at winstone.HttpListener.parseURILine(HttpListener.java:318)<br/>
	at winstone.HttpListener.parseURI(HttpListener.java:246)<br/>
	at winstone.RequestHandlerThread.run(RequestHandlerThread.java:87)<br/>
	at java.lang.Thread.run(Thread.java:619)</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4809 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4810 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>I'm not able to reproduce this issue, but it has been filed by 3 people.. can<br/>
you all ensure your browser cache is clear and see if that helps.  If not,<br/>
please post your environment details so we can find what common factors might<br/>
cause this problem..  OS type, java version, webserver type/version, browser<br/>
type/version, list of Hudson plugins installed.  Thanks!</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>Ok, I have been able to reproduce the problem now.. investigating.</p>

<p>(for some reason, running a development environment from maven, "mvn<br/>
hudson-dev:run", does not show the problem.. but using "java -jar hudson.war" I<br/>
do see it)</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>Ok, the problem is from a change added into Winstone in 1.333.  Kohsuke will fix<br/>
it for 1.334.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>Confirmed the problem to be in Winstone.</p>

<p>The fix in Winstone is in<br/>
<a href="http://kenai.com/projects/hudson/sources/winstone/revision/49" class="external-link" target="_blank" rel="nofollow noopener">http://kenai.com/projects/hudson/sources/winstone/revision/49</a> (changeset<br/>
dfeff50e1e3c)</p>

<p>Targeted for integration in 1.334.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : kohsuke<br/>
Path:<br/>
 trunk/hudson/main/war/pom.xml<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=23635" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=23635</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-4804&#93;</span> Integrated a fix in Winstone towrad 1.334</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hans23">hans23</a>:</i>
<p>Is there any way to get this fix into an existing installation.  Or, <br/>
alternatively, are there any plans to get 1.334 out soon?  My hudson installation <br/>
is basically blocked by this bug.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>hans23, the easiest option is to rollback to 1.332 until 1.334 is out (probably<br/>
friday).  But, you could also try replacing the winstone jar with the new one:<br/>
<a href="http://download.java.net/maven/2/org/jvnet/hudson/winstone/winstone/0.9.10-hudson-16/" class="external-link" target="_blank" rel="nofollow noopener">http://download.java.net/maven/2/org/jvnet/hudson/winstone/winstone/0.9.10-hudson-16/</a></p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hans23">hans23</a>:</i>
<p>Downgrading to 1.332 worked well, thanks for responding!</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4818 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4824 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4835 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 4841 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tonioska">tonioska</a>:</i>
<p>Hi kohsuke,<br/>
First of all, thanks for this brilliant application, to you and all the people behind it.</p>

<p>Now, im using version 1.339 and i get the following error every time i go to Manage Hudson -> Manage Plugins, resulting in blank Updates and Available tabs :</p>

<p>java.lang.IllegalStateException: Form too large218763>200000<br/>
at org.mortbay.jetty.Request.extractParameters(Request.java:1276)<br/>
at org.mortbay.jetty.Request.getParameterNames(Request.java:664)<br/>
at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:59)<br/>
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1084)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:166)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.basicauth.BasicProcessingFilter.doFilter(BasicProcessingFilter.java:173)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)<br/>
at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:66)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1084)<br/>
at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:360)<br/>
at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)<br/>
at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:181)<br/>
at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:712)<br/>
at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:405)<br/>
at org.mortbay.jetty.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:211)<br/>
at org.mortbay.jetty.handler.HandlerCollection.handle(HandlerCollection.java:114)<br/>
at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:139)<br/>
at org.mortbay.jetty.Server.handle(Server.java:313)<br/>
at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:506)<br/>
at org.mortbay.jetty.HttpConnection$RequestHandler.content(HttpConnection.java:844)<br/>
at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:644)<br/>
at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:205)<br/>
at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:381)<br/>
at org.mortbay.io.nio.SelectChannelEndPoint.run(SelectChannelEndPoint.java:396)<br/>
at org.mortbay.thread.BoundedThreadPool$PoolThread.run(BoundedThreadPool.java:442)</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>tonioska, how do you run Hudson?  The problem should be resolved for "java -jar hudson.war" but if you run in your own jetty container then maybe your jetty configuration needs to be updated.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>tonioska, I am reclosing this issue because your case is not related.<br/>
There were some changes related to this Jetty limitation, though I don't see that any issue was filed.  Here are the changes:<br/>
<a href="https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24378" class="external-link" target="_blank" rel="nofollow noopener">https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24378</a><br/>
<a href="https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24390" class="external-link" target="_blank" rel="nofollow noopener">https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24390</a><br/>
<a href="https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24469" class="external-link" target="_blank" rel="nofollow noopener">https://hudson.dev.java.net/source/browse/hudson?view=rev&rev=24469</a></p>

<p>Hopefully you can find a jetty configuration to do the same as the above changes.  If so, please put it on our wiki:<br/>
<a href="http://wiki.jenkins-ci.org/display/JENKINS/Jetty" class="external-link" target="_blank" rel="nofollow noopener">http://wiki.jenkins-ci.org/display/JENKINS/Jetty</a><br/>
Thanks!</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vbeffa">vbeffa</a>:</i>
<p>My problem is more closely related to <a href="https://issues.jenkins-ci.org/browse/JENKINS-4833" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-4833</a> but that is marked as a duplicate of this bug.</p>

<p>I just installed Jenkins 1.502 on RedHat CentOS 6.3. Because the server already has Apache on it I set JENKINS_PORT="8081" and JENKINS_AJP_PORT="-1" to avoid port conflicts. Jenkins starts up OK. When I go to Manage Plugins, though, the list is empty. In the log file I see the same exception described above.</p>

<p>I tried to update the site on the Advanced tab using the default URL <a href="http://updates.jenkins-ci.org/update-center.json" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins-ci.org/update-center.json</a> but that didn't help.</p>

<p>WARNING: Untrapped servlet exception<br/>
javax.servlet.ServletException: net.sf.json.JSONException: Unterminated string at character 228525 of {"connectionCheckUrl": "http://www.google.com/", "core"<br/>
...<br/>
8-22T16:03:14.00Z", "requiredCore": "1.466", "scm": "github.com", "sha1": "uWryTtC4mZPwLfmZ8JjHHT1tZDo=", "title": "Groovy Postbuild Plugin", "url": "http://updates.jenkins-ci.org/download/plu<br/>
at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:615)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:658)<br/>
at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:241)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)<br/>
at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:573)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:658)<br/>
at org.kohsuke.stapler.MetaClass$4.doDispatch(MetaClass.java:203)<br/>
at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)<br/>
at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:573)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:658)<br/>
at org.kohsuke.stapler.Stapler.invoke(Stapler.java:487)<br/>
at org.kohsuke.stapler.Stapler.service(Stapler.java:164)<br/>
at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)<br/>
at winstone.ServletConfiguration.execute(ServletConfiguration.java:248)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:333)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:376)<br/>
at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:95)<br/>
at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:87)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)<br/>
at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:47)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:124)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.ui.basicauth.BasicProcessingFilter.doFilter(BasicProcessingFilter.java:174)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at jenkins.security.ApiTokenFilter.doFilter(ApiTokenFilter.java:64)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)<br/>
at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:66)<br/>
at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)<br/>
at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)<br/>
at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:50)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)<br/>
at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:81)<br/>
at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)<br/>
at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)<br/>
at winstone.RequestDispatcher.forward(RequestDispatcher.java:331)<br/>
at winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:227)<br/>
at winstone.RequestHandlerThread.run(RequestHandlerThread.java:150)<br/>
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)<br/>
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)<br/>
at java.util.concurrent.FutureTask.run(FutureTask.java:166)<br/>
at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)<br/>
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1146)<br/>
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)<br/>
at java.lang.Thread.run(Thread.java:679)</p>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4824">JENKINS-4824</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4809">JENKINS-4809</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4818">JENKINS-4818</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4833">JENKINS-4833</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4835">JENKINS-4835</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4841">JENKINS-4841</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4997">JENKINS-4997</a>]</i>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-4810">JENKINS-4810</a>]</i>
